src/ts/interaction.ts

```diff
@@ -11,30 +11,41 @@ export interface Interaction {
   /**
    * Call value in wei for the interaction, allowing Ether to be sent.
    */
-  value?: BigNumberish;
+  value: BigNumberish;
   /**
    * Call data used in the interaction with a smart contract.
    */
-  callData?: BytesLike;
+  callData: BytesLike;
 }
 
-export function encodeInteraction(interaction: Interaction): string {
-  const value = BigNumber.from(interaction.value || 0);
-  const callData = interaction.callData || "0x";
-  const callDataLength = ethers.utils.hexDataLength(callData);
+export type InteractionLike = Pick<Interaction, "target"> &
+  Partial<Interaction>;
 
-  const encodedInteraction = value.isZero()
-    ? ethers.utils.solidityPack(
-        ["address", "bool", "uint24", "bytes"],
-        [interaction.target, false, callDataLength, callData],
-      )
-    : ethers.utils.solidityPack(
-        ["address", "bool", "uint24", "uint256", "bytes"],
-        [interaction.target, true, callDataLength, value, callData],
-      );
-  return encodedInteraction;
+/**
+ * Normalizes interaction data so that it is ready to be be ABI encoded.
+ *
+ * @param interaction The interaction to normalize.
+ * @return The normalized interaction.
+ */
+export function normalizeInteraction(
+  interaction: InteractionLike,
+): Interaction {
+  return {
+    value: 0,
+    callData: "0x",
+    ...interaction,
+  };
```

Comment:
Did we have this before, why is it only now introduced? I guess because we are passing a struct now instead of bytes to the contract, so we need to transform it back into appropriate bytes.

Reply:
We did, its about setting optional interaction options to the default values.